File: src/KI_36/Proshkin/Lab3/Bullet.java
package KI_36.Proshkin.Lab3;

/**
 * Клас, що представляє боєприпаси (Patron) для зброї.
 */
public class Bullet {
    private String caliber;
    private int quantity;

    public Bullet(String caliber, int quantity) {
        this.caliber = caliber;
        this.quantity = quantity;
    }

    public String getCaliber() {
        return caliber;
    }

    public int getQuantity() {
        return quantity;
    }

    public void use() {
        if (quantity > 0) {
            quantity--;
        }
    }

    public void add(int quantity) {
        this.quantity += quantity;
    }

    @Override
    public String toString() {
        return "Bullet{" +
                "caliber='" + caliber + '\'' +
                ", quantity=" + quantity +
                '}';
    }
}

File: Bullet.py
class Bullet:
    def __init__(self, caliber, quantity):
        self.caliber = caliber
        self.quantity = quantity

    def get_caliber(self):
        return self.caliber

    def get_quantity(self):
        return self.quantity

    def use(self):
        if self.quantity > 0:
            self.quantity -= 1

    def add(self, quantity):
        self.quantity += quantity

    def __str__(self):
        return f"Bullet(caliber='{self.caliber}', quantity={self.quantity})"